### ISSUE TITLE
使用三种训练模型ch_ppocr_xxx_v2.0_rec， inter_rec都不对

### ISSUE THREAD

**Issue by suchxyz** [PaddlePaddle/PaddleOCR#5352]
请提供下述完整信息以便快速定位问题/Please provide the following information to quickly locate the problem

- 系统环境/System Environment：windows
- 版本号/Version：Paddle：2.2.1   PaddleOCR：2.4  问题相关组件/Related components： tools/infer_rec
- 运行指令/Command Code：
tools/infer_rec.py -c configs/rec/ch_ppocr_v2.0/rec_chinese_lite_train_v2.0_cpu.yml -o Global.pretrained_model=pretrain_models/ch_ppocr_mobile_v2.0_rec_train/best_accuracy Global.load_static_weights=false Global.infer_img=doc/imgs_words/ch/word_1.jpg

- 完整报错/Complete Error Message：
https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.4/doc/doc_ch/models_list.md
按照上面这个文档里的连接下的训练模型， 分别用了ch_ppocr_mobile_slim_v2.0_rec、ch_ppocr_mobile_v2.0_rec、ch_ppocr_server_v2.0_rec 和对应的配置文件，试了infer_rec.py 都不对

**[2022/01/25 18:36:17] root WARNING: The shape of model params head.fc.weight [96, 6627] not matched with loaded params head.fc.weight [96, 6625] !**
**[2022/01/25 18:36:17] root WARNING: The shape of model params head.fc.bias [6627] not matched with loaded params head.fc.bias [6625] !**
[2022/01/25 18:36:17] root INFO: load pretrain successful from pretrain_models/ch_ppocr_mobile_v2.0_rec_train/best_accuracy
[2022/01/25 18:36:17] root INFO: infer_img: doc/imgs_words/ch/word_1.jpg
**[2022/01/25 18:36:17] root INFO: 	 result: 尬棍妗诺喜５瀑痧群瀑Ω,嗔當恐尬帜尬寻践妗程意飼践醚践还	0.00046281534**
[2022/01/25 18:36:17] root INFO: success!

配置文件：
Global:
  use_gpu: false
  epoch_num: 500
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: ./output/rec_chinese_lite_v2.0
  save_epoch_step: 3
  # evaluation is run every 5000 iterations after the 4000th iteration
  eval_batch_step: [0, 2000]
  cal_metric_during_train: True
  pretrained_model:
  checkpoints: 
  save_inference_dir:
  use_visualdl: False
  infer_img: doc/imgs_words/ch/word_1.jpg
  # for data or label process
  character_dict_path: ppocr/utils/ppocr_keys_v1.txt
  max_text_length: 25
  infer_mode: False
  use_space_char: True
  save_res_path: ./output/rec/predicts_chinese_lite_v2.0.txt

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    name: Cosine
    learning_rate: 0.001
    warmup_epoch: 5
  regularizer:
    name: 'L2'
    factor: 0.00001

Architecture:
  model_type: rec
  algorithm: CRNN
  Transform:
  Backbone:
    name: MobileNetV3
    scale: 0.5
    model_name: small
    small_stride: [1, 2, 2, 2]
  Neck:
    name: SequenceEncoder
    encoder_type: rnn
    hidden_size: 48
  Head:
    name: CTCHead
    fc_decay: 0.00001

Loss:
  name: CTCLoss

PostProcess:
  name: CTCLabelDecode

Metric:
  name: RecMetric
  main_indicator: acc

Train:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data/
    label_file_list: ["./train_data/train_list.txt"]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - RecAug: 
      - CTCLabelEncode: # Class handling label
      - RecResizeImg:
          image_shape: [3, 32, 320]
      - KeepKeys:
          keep_keys: ['image', 'label', 'length'] # dataloader will return list in this order
  loader:
    shuffle: True
    batch_size_per_card: 256
    drop_last: True
    num_workers: 8

Eval:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data
    label_file_list: ["./train_data/val_list.txt"]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - CTCLabelEncode: # Class handling label
      - RecResizeImg:
          image_shape: [3, 32, 320]
      - KeepKeys:
          keep_keys: ['image', 'label', 'length'] # dataloader will return list in this order
  loader:
    shuffle: False
    drop_last: False
    batch_size_per_card: 256
    num_workers: 8

**Comment by suchxyz** [comment]
另外试了rec_mv3_none_bilstm_ctc_v2.0_train这个英文模型，结果是对的。

-c configs/rec/rec_mv3_none_bilstm_ctc.yml -o Global.pretrained_model=pretrain_models/rec_mv3_none_bilstm_ctc_v2.0_train/best_accuracy Global.load_static_weights=false Global.infer_img=doc/imgs_words/en/word_1.png

[2022/01/26 10:22:48] root INFO: train with paddle 2.2.1 and device CPUPlace
[2022/01/26 10:22:48] root INFO: load pretrain successful from pretrain_models/rec_mv3_none_bilstm_ctc_v2.0_train/best_accuracy
[2022/01/26 10:22:48] root INFO: infer_img: doc/imgs_words/en/word_1.png
[2022/01/26 10:22:48] root INFO: 	 result: joint	0.9999008
[2022/01/26 10:22:48] root INFO: success!

**Comment by tink2123** [comment]
```
[2022/01/25 18:36:17] root WARNING: The shape of model params head.fc.weight [96, 6627] not matched with loaded params head.fc.weight [96, 6625] !
[2022/01/25 18:36:17] root WARNING: The shape of model params head.fc.bias [6627] not matched with loaded params head.fc.bias [6625] !
```

看报错，模型权重没有被正确加载上，请问是否修改了字典呢？

**Comment by suchxyz** [comment]
改了字典，加了两个字符。 去掉就可以了。
不过还有个问题， train的时候， 必须要加那两个字符，开始的时候也报这两个warning，会有影响吗？

**Comment by suchxyz** [comment]
> ```
> [2022/01/25 18:36:17] root WARNING: The shape of model params head.fc.weight [96, 6627] not matched with loaded params head.fc.weight [96, 6625] !
> [2022/01/25 18:36:17] root WARNING: The shape of model params head.fc.bias [6627] not matched with loaded params head.fc.bias [6625] !
> ```
> 
> 看报错，模型权重没有被正确加载上，请问是否修改了字典呢？

改了字典，加了两个字符。 去掉就可以了。多谢多谢！

不过还有个问题， train的时候， 必须要加那两个字符，开始的时候也报这两个warning，会有影响吗？

**Comment by suchxyz** [comment]
> 



> > ```
> > [2022/01/25 18:36:17] root WARNING: The shape of model params head.fc.weight [96, 6627] not matched with loaded params head.fc.weight [96, 6625] !
> > [2022/01/25 18:36:17] root WARNING: The shape of model params head.fc.bias [6627] not matched with loaded params head.fc.bias [6625] !
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 看报错，模型权重没有被正确加载上，请问是否修改了字典呢？
> 
> 改了字典，加了两个字符。 去掉就可以了。多谢多谢！
> 
> 不过还有个问题， train的时候， 必须要加那两个字符，开始的时候也报这两个warning，会有影响吗？


加了两个字符训练了以后， inter_rec.py了一下，发现所有result都是那两个字符了...
我训练数据只添加了，这两个字符相关的，因为这两个字符的场景都是单独出现的。 
是不是训练的时候要添加全量所有中文的数据？ 能做到只训练新字符的数据，但不影响原来其他中文字符识别的效果吗？

**Comment by ziyaxuanyi** [comment]
最好是加全量数据，把你需要识别的所有字符做成一个字典，然后构建包含这个字典里字符的数据集，保证字典里面每个字符都要出现且最好分布均匀。你自己学习实验的话，数据量在几万到几十万就可以了，但如果是要实际落地部署使用，那数据量要达到千万级别才能有比较好的效果。

只训练新字符的数据，但不影响原来其他中文字符识别的效果的话，可以尝试以较小的学习率去微调，而且固定网络的大部分参数，只训练最后几层的参数，但前提是这些字符必须包含在官方模型对应的字典里。如果你要训练的新字符不在官方字典里面，那就只能重新构建字典和数据集训练了，这个训练可以加载官方预训练模型。

**Comment by suchxyz** [comment]
多谢！

**Comment by xuxiansheng2018** [comment]
> 最好是加全量数据，把你需要识别的所有字符做成一个字典，然后构建包含这个字典里字符的数据集，保证字典里面每个字符都要出现且最好分布均匀。你自己学习实验的话，数据量在几万到几十万就可以了，但如果是要实际落地部署使用，那数据量要达到千万级别才能有比较好的效果。
> 
> 只训练新字符的数据，但不影响原来其他中文字符识别的效果的话，可以尝试以较小的学习率去微调，而且固定网络的大部分参数，只训练最后几层的参数，但前提是这些字符必须包含在官方模型对应的字典里。如果你要训练的新字符不在官方字典里面，那就只能重新构建字典和数据集训练了，这个训练可以加载官方预训练模型。

如果我修改字典 替换了一个字 训练的时候 还能拿着官方的预训练模型 后面接着训练吗？    我现在的识别效果不好 感觉是我修改了字典的问题 大佬有什么建议吗
